draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+## JRI Keywords
+
+JRI defines keywords in three behavior categories:
+
+* Identifiers, which assign IRIs to primary or secondary resources and (for primary resources) set the resource's base IRI
+* Locations, which indicate which locations are referenceable and can contain identifiers
+* References, which link one part of a document to another part of the same document or to a different document
+
+Context specifications MAY define their own keywords or other mechanisms within these three categories, as discussed in {{context}}.  They also MAY restrict reference targets to those identified by JRI and/or context specification identifier or location keywords.
+
+### General syntax
+
+JRI keyword names and string values are defined to be representable as {{?RFC3629}} UTF-8 strings, with keyword names representable in {{?RFC20}} ASCII.  As such, JRI keywords can be used as object properties in interoperable JSON documents as prescribed by {{?RFC8259, Sections 4, 7, and 8.1}}, as well as in any other JSON Pointer-compatible data formats.
+
+JRI keywords MAY be used in data formats that are only capable of representing a subset of UTF-8 string values, with the obvious limitation that only the representable subset of JRI keyword values will be usable.
+
+As {{!RFC3987, Section 3}} defines mappings between IRI and URI syntax, JRI MAY be used in data formats that can only directly represent URIs.
+
+#### The "$" prefix
+
+JRI keywords MUST start with a "$" character.  Context specifications MAY define their own "$"-prefixed keywords, however this is NOT RECOMMENDED as future specifications that extend JRI as discussed in {{extending}} will use that prefix.[^8]
+
+[^8]: The usage of "$" as a prefix in the JSON Schema Core Vocabulary pre-dates JRI, and is expected to continue.  Some older JSON Schema proposals from immediately post-draft-04 also used a "$" prefix, but none were adopted and JSON Schema now discourages the use of that prefix.
+
+#### Exclusion of "$comment"
+
+For compatibility with JSON Schema {{?I-D.bhutton-json-schema-01, Section 8.3}}, JRI requires that a `"$comment"` keyword present in the same object as any JRI keywords MUST NOT impact the semantics of those JRI keywords.  In particular, `"$comment"` MUST NOT be used as a semantic disambiguator as described in {{disambiguator}}.
+
+However, `"$comment"` is not itself considered to be a JRI keyword.  JRI does not place any requirements on the usage, data type, syntax, or other semantics of `"$comment"` in context specifications.  JRI implementations MUST NOT require that a `"$comment"` adjacent to JRI keywords conform to JSON Schema's `"$comment"` keyword, unless explicitly required to do so by a context specification.
```

Comment:
@DavidBiesack JSON Schema's `$comment` keyword is bizarrely controversial.  I don't understand it at all, but some people really hate it.  I'm kind of trying to make sure it doesn't get stepped on without causing the anti-`$comment` brigade to come after me.

I would have no objection to including it, but I know there would be objections.  It might be something that has to get sorted out in an issue as I doubt we could sort it out within this PR in a timely manner.

Reply:
See #9.  And to some degree, #10.